// angular2-tree-component/dist/directives/tree-node-drop.directive.js
"use strict";
var core_1 = require('@angular/core');
var tree_node_drag_model_1 = require('../models/tree-node-drag.model');
var DRAG_OVER_CLASS = 'is-dragging-over';
var TreeNodeDrop = (function () {
    function TreeNodeDrop(el, renderer, treeNodeDrag) {
        this.el = el;
        this.renderer = renderer;
        this.treeNodeDrag = treeNodeDrag;
    }
    TreeNodeDrop.prototype.onDragOver = function ($event) {
        $event.preventDefault();
        this.addClass();
    };
    TreeNodeDrop.prototype.onDragLeave = function ($event) {
        var rect = this.el.nativeElement.getBoundingClientRect();
        // If outside the element
        if ($event.clientX < rect.left || $event.clientX > rect.right ||
            $event.clientY < rect.top || $event.clientY > rect.bottom) {
            this.removeClass();
        }
    };
    TreeNodeDrop.prototype.onDrop = function ($event) {
        $event.preventDefault();
        this.treeNodeDrop({ event: $event, node: this.treeNodeDrag.getDragNode() });
        this.removeClass();
    };
    TreeNodeDrop.prototype.addClass = function () {
        this.renderer.setElementClass(this.el.nativeElement, DRAG_OVER_CLASS, true);
    };
    TreeNodeDrop.prototype.removeClass = function () {
        this.renderer.setElementClass(this.el.nativeElement, DRAG_OVER_CLASS, false);
    };
    TreeNodeDrop.decorators = [
        { type: core_1.Directive, args: [{
                    selector: '[treeNodeDrop]'
                },] },
    ];
    /** @nocollapse */
    TreeNodeDrop.ctorParameters = [
        { type: core_1.ElementRef, },
        { type: core_1.Renderer, },
        { type: tree_node_drag_model_1.TreeNodeDrag, },
    ];
    TreeNodeDrop.propDecorators = {
        'treeNodeDrop': [{ type: core_1.Input, args: ['treeNodeDrop',] },],
        'onDragOver': [{ type: core_1.HostListener, args: ['dragover', ['$event'],] },],
        'onDragLeave': [{ type: core_1.HostListener, args: ['dragleave', ['$event'],] },],
        'onDrop': [{ type: core_1.HostListener, args: ['drop', ['$event'],] },],
    };
    return TreeNodeDrop;
}());
exports.TreeNodeDrop = TreeNodeDrop;
