// angular2-tree-component/dist/models/tree-node-drag.model.js
"use strict";
var core_1 = require('@angular/core');
var TreeNodeDrag = (function () {
    function TreeNodeDrag() {
        this._dragNode = null;
        this._dropLocation = null;
    }
    // TODO: move to a different service:
    TreeNodeDrag.prototype.setDragNode = function (dragNode) {
        this._dragNode = dragNode;
    };
    TreeNodeDrag.prototype.getDragNode = function () {
        return this._dragNode || { node: null, index: null };
    };
    TreeNodeDrag.prototype.isDragging = function () {
        return this.getDragNode().node;
    };
    TreeNodeDrag.prototype.setDropLocation = function (dropLocation) {
        this._dropLocation = dropLocation;
    };
    TreeNodeDrag.prototype.getDropLocation = function () {
        return this._dropLocation || { component: null, node: null, index: null };
    };
    TreeNodeDrag.prototype.isDraggingOver = function (component) {
        return this.getDropLocation().component === component;
    };
    TreeNodeDrag.prototype.endDrag = function () {
        this.setDropLocation(null);
        this.setDragNode(null);
    };
    TreeNodeDrag.decorators = [
        { type: core_1.Injectable },
    ];
    /** @nocollapse */
    TreeNodeDrag.ctorParameters = [];
    return TreeNodeDrag;
}());
exports.TreeNodeDrag = TreeNodeDrag;
