// angular2-tree-component/dist/angular2-tree-component.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var tree_options_model_1 = require("./models/tree-options.model");
exports.TREE_ACTIONS = tree_options_model_1.TREE_ACTIONS;
var keys_1 = require("./constants/keys");
exports.KEYS = keys_1.KEYS;
var tree_model_1 = require("./models/tree.model");
exports.TreeModel = tree_model_1.TreeModel;
var tree_node_model_1 = require("./models/tree-node.model");
exports.TreeNode = tree_node_model_1.TreeNode;
var tree_dragged_element_model_1 = require("./models/tree-dragged-element.model");
exports.TreeDraggedElement = tree_dragged_element_model_1.TreeDraggedElement;
var loading_component_1 = require("./components/loading.component");
exports.LoadingComponent = loading_component_1.LoadingComponent;
var deprecated_loading_component_1 = require("./components/deprecated-loading.component");
var tree_component_1 = require("./components/tree.component");
exports.TreeComponent = tree_component_1.TreeComponent;
var tree_node_component_1 = require("./components/tree-node.component");
exports.TreeNodeComponent = tree_node_component_1.TreeNodeComponent;
var tree_node_content_component_1 = require("./components/tree-node-content.component");
exports.TreeNodeContent = tree_node_content_component_1.TreeNodeContent;
var deprecated_tree_node_content_component_1 = require("./components/deprecated-tree-node-content.component");
var tree_node_drop_slot_component_1 = require("./components/tree-node-drop-slot.component");
exports.TreeNodeDropSlot = tree_node_drop_slot_component_1.TreeNodeDropSlot;
var tree_drop_directive_1 = require("./directives/tree-drop.directive");
exports.TreeDropDirective = tree_drop_directive_1.TreeDropDirective;
var tree_drag_directive_1 = require("./directives/tree-drag.directive");
exports.TreeDragDirective = tree_drag_directive_1.TreeDragDirective;
var adhoc_component_factory_service_1 = require("./components/adhoc-component-factory.service");
require("./polyfills");
var deprecated_1 = require("./deprecated");
var TreeModule = (function () {
    function TreeModule() {
    }
    return TreeModule;
}());
TreeModule = __decorate([
    core_1.NgModule({
        declarations: [
            loading_component_1.LoadingComponent,
            tree_component_1.TreeComponent,
            tree_node_component_1.TreeNodeComponent,
            tree_node_drop_slot_component_1.TreeNodeDropSlot,
            tree_node_content_component_1.TreeNodeContent,
            tree_drop_directive_1.TreeDropDirective,
            tree_drag_directive_1.TreeDragDirective
        ],
        exports: [
            tree_component_1.TreeComponent,
            tree_drop_directive_1.TreeDropDirective,
            tree_drag_directive_1.TreeDragDirective
        ],
        imports: [
            common_1.CommonModule,
        ],
        providers: [
            tree_dragged_element_model_1.TreeDraggedElement
        ]
    }),
    __metadata("design:paramtypes", [])
], TreeModule);
exports.TreeModule = TreeModule;
var DeprecatedTreeModule = (function () {
    function DeprecatedTreeModule() {
        deprecated_1.deprecated('DeprecatedTreeModule', 'TreeModule for AoT compilation');
    }
    return DeprecatedTreeModule;
}());
DeprecatedTreeModule = __decorate([
    core_1.NgModule({
        declarations: [
            deprecated_loading_component_1.LoadingComponent,
            deprecated_tree_node_content_component_1.TreeNodeContent
        ],
        exports: [
            tree_component_1.TreeComponent,
            tree_drop_directive_1.TreeDropDirective,
            tree_drag_directive_1.TreeDragDirective
        ],
        imports: [
            common_1.CommonModule,
            TreeModule,
        ],
        providers: [
            adhoc_component_factory_service_1.AdHocComponentFactoryCreator
        ],
    }),
    __metadata("design:paramtypes", [])
], DeprecatedTreeModule);
exports.DeprecatedTreeModule = DeprecatedTreeModule;
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = TreeModule;
